### ISSUE TITLE
Building FunctionalTest and Sample App on iOS fails (asserts in cross compiler) when using LLVM (simulator x64).

### ISSUE THREAD

**Issue by lateralusX** [dotnet/runtime#51992]
Building functional test targeting iOS simulator fails when using full AOT LLVM, not using LLVM works as expected.

Initial error is an assert when compiling S.P.C:

* Assertion at /Users/jolorens/github/runtime/src/mono/mono/mini/simd-intrinsics.c:774, condition `cfg->r4fp' not met (TaskId:672)

This seems to happen since iOS disables float32 by default:

https://github.com/dotnet/runtime/blob/3bf6e21ccab8705282e5cdebee6998f36ac1080b/src/tasks/AotCompilerTask/MonoAOTCompiler.props#L14

and above hits an intrinsics that excepts that float32 optimisation is enabled.

If that issue is fixed, S.P.C succeed compiling, but fails to link app, just hangs (or taking very long time to link).

When I build sample app using LLVM I got errors in the generated assembler files, giving errors like this:

Users/jolorens/github/runtime/src/mono/sample/iOS/bin/iossimulator-x64/publish/System.Security.Principal.Windows.dll.s:127:1: error: file number already allocated
.file 1 "<unknown>"

Building full AOT without AOT using Debug configuration fails with same error as above.

**Comment by SamMonoRT** [comment]
cc @imhameed - please take a look

**Comment by lateralusX** [comment]
Building using xcodebuild release configuration won't hit the .file 1 error (because release build won't use that info).

**Comment by vargaz** [comment]
Debug mode + llvm is not going to work, llvm compiled code doesn't support debugging.

**Comment by lateralusX** [comment]
@vargaz This was native debug build, and the issue is in assembler file emitted by JIT AOT compiler, so exists in native debug builds for none llvm build as well. I have build native debug build with LLVM generated objects on other platforms without issues. I know we don’t support managed debugging on LLVM, or did you refer to something else?

**Comment by steveisok** [comment]
@imhameed @vargaz Should the guidance be to remove `-O=-float32` for now?

**Comment by vargaz** [comment]
Yes.

**Comment by steveisok** [comment]
> If that issue is fixed, S.P.C succeed compiling, but fails to link app, just hangs (or taking very long time to link).

I can reproduce if you add `<MonoEnableLLVM>true</MonoEnableLLVM>` to the AOT functional test (also noticed it was missing).  

https://github.com/dotnet/runtime/blob/main/src/tests/FunctionalTests/iOS/Simulator/AOT/iOS.Simulator.Aot.Test.csproj

**Comment by SamMonoRT** [comment]
@directhex - while investigating https://github.com/dotnet/runtime/issues/54572, did you encounter this issue, and/or know who the right person is to investigate this ?

**Comment by steveisok** [comment]
@SamMonoRT https://github.com/dotnet/runtime/issues/54572 was intended to address the sample part of this issue.  We should be good.

**Comment by imhameed** [comment]
I can build and run the AOT and AOT-LLVM functional tests. Closing.

For posterity, I built the runtime with:
`./build.sh mono+libs -os iOSSimulator -arch x64 -configuration Release`

And I built the AOT-LLVM functional test with:
`./dotnet.sh build /t:Test /p:TargetOS=iOSSimulator /p:TargetArchitecture=x64 /p:Configuration=Release src/tests/FunctionalTests/iOS/Simulator/AOT-LLVM`

I did have to run a locally-built copy of xharness manually to actually run the app in the simulator, though, and the app's exit code wasn't communicated properly back to xharness.
